src/librustdoc/formats/cache.rs

```diff
@@ -131,44 +130,21 @@ struct CacheBuilder<'a, 'tcx> {
 }
 
 impl Cache {
-    crate fn from_krate<'tcx>(
-        render_info: RenderInfo,
-        document_private: bool,
+    crate fn new(access_levels: AccessLevels<DefId>, document_private: bool) -> Self {
```

Comment:
Could you explain to me a bit why `access_levels` and `document_private` are initialized using `::new()` but everything else is initialized directly by the users using assignment?

Reply:
I think because they're already available when the cache is constructed, but the other fields are set later.

Reply:
Seems fine.